### DISCUSSION TITLE
How to get binary date from S3?

### DISCUSSION THREAD

**Discussion by tonal** [PrefectHQ/prefect#4226] [General] [answered]
How to get binary date from S3?
Such as archive files, xls, dbf and similar.

**Comment by zanieb** [comment, marked as answer]
Hi! https://docs.prefect.io/api/latest/tasks/aws.html#s3download supports `gzip` archives but it does look like the `.decode()` method is called at the end of the method which may cause problems but I presume you can just `.encode()` the result to bytes again then do what you want? If not, we'd be happy to accept a PR (or open an issue) to add support for returning bytes directly.

You can also use that task as a template for writing your own task that does what you need, the source is at https://github.com/PrefectHQ/prefect/blob/master/src/prefect/tasks/aws/s3.py#L11

**Reply by tonal** [reply]
https://github.com/PrefectHQ/prefect/pull/4238
